seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/serialization/InternalRowCollector.java

```diff
@@ -50,7 +48,7 @@ public InternalRowCollector(
         this.rowSerialization = new InternalRowConverter(dataType);
         this.collectTotalCount = new AtomicLong(0);
         this.envOptions = (Map) envOptionsInfo;
-        FlowControlStrategy flowControlStrategy = getFlowControlStrategy(envOptions);
+        FlowControlStrategy flowControlStrategy = FlowControlStrategy.fromMap(envOptions);
```

Comment:
After this changed, the `FlowControlStrategy` always be not null, so even user not configure flow control, some method will be invoked in https://github.com/apache/seatunnel/blob/d4a323efef3979c44cda20cc4b6c7b04b7454f5a/seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/serialization/InternalRowCollector.java#L63

Reply:
I have a problem, why design this can be null and setting default value (Integer.MAX_VALUE) in flow control strategy? If user dose not set these values, How to reflect speed limit behavior? In the implementation of the three engines, this is completely inconsistent. Where is the detailed STIP for this feature?

Reply:
> why design this can be null 

null means not enabling this feature, so we can save unnecessary method calls.

> and setting default value (Integer.MAX_VALUE) in flow control strategy

default value as `Integer.MAX_VALUE` only worked on only one of the number of rows or bytes set by the user will take effect. In fact, `Integer.MAX_VALUE` basically represents unlimited. eg: user config bytes == 10, so we will set rows count as `Integer.MAX_VALUE`. If we feel there is something wrong with the current code, we can improve it.

> In the implementation of the three engines, this is completely inconsistent.

Which part？

> Where is the detailed STIP for this feature?

I personally feel that the amount of changes to this function is very small, and it is also a small function with little impact. Not all feature require STIP. If you think this feature requires a STIP, please let me know when created the PR.

Reply:
I think rate limit need a complete design document is needed to clearly define the boundaries of the functionality. I will close this pull request and open again after discussing and implementing the plan.